pkg/apis/v1alpha1/awsnodetemplate.go

```diff
@@ -15,36 +15,51 @@ limitations under the License.
 package v1alpha1
 
 import (
+	v1 "k8s.io/api/core/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
-// SubnetStatus contains resolved Subnet selector values utilized for node launch
-type SubnetStatus struct {
+// Subnet contains resolved Subnet selector values utilized for node launch
+type Subnet struct {
 	// Id of the subnet
-	// +optional
-	ID string `json:"id,omitempty"`
+	// +required
+	ID string `json:"id"`
 	// The associated availability zone
-	// +optional
-	Zone string `json:"zone,omitempty"`
+	// +required
+	Zone string `json:"zone"`
 }
 
-// SecurityGroupStatus contains resolved SecurityGroup selector values utilized for node launch
-type SecurityGroupStatus struct {
+// SecurityGroup contains resolved SecurityGroup selector values utilized for node launch
+type SecurityGroup struct {
 	// Id of the security group
-	// +optional
-	ID string `json:"id,omitempty"`
+	// +required
+	ID string `json:"id"`
+}
+
+// AMI contains resolved AMI selector values utilized for node launch
+type AMI struct {
+	// Id of the AMI
```

Comment:
What do you think about including the AMI name? This generally has a more meaningful name for the user than just the id?